src/server/actions/getGeneratedSmartAccount.ts

```diff
@@ -0,0 +1,116 @@
+"use server"
+
+import { LocalAccountSigner } from "@alchemy/aa-core"
+import { LinkedAccountWithMetadata, PrivyClient, WalletWithMetadata } from "@privy-io/server-auth"
+import { eq, isNotNull, or } from "drizzle-orm"
+import { and } from "drizzle-orm/sql/expressions/conditions"
+import { cookies } from "next/headers"
+import { Address, Hex } from "viem"
+import { getLightAccountAddress } from "@/server/actions/lib/lightAccount"
+import { db, schema } from "@/server/db/db"
+import { activeChain } from "@/utils/chain"
+import { createSmartAccountClient } from "@/utils/userOperation"
+
+const privy = new PrivyClient(process.env.NEXT_PUBLIC_PRIVY_APP_ID!, process.env.PRIVY_APP_SECRET!)
+
+const isWalletWithMetadata = (account: LinkedAccountWithMetadata): account is WalletWithMetadata =>
+  account.type === "wallet" && account.walletClientType === "privy"
+
+export const getGeneratedSmartAccount = async () => {
+  const authToken = cookies().get("privy-token")?.value
+
+  if (authToken === undefined) {
+    console.error("Auth token not found")
+    return undefined
+  }
+
+  const claims = await privy.verifyAuthToken(authToken)
+  const user = await privy.getUser(claims.userId)
+
+  const walletAddress = user.linkedAccounts.find(isWalletWithMetadata)?.address as Address
+
+  if (walletAddress === undefined) {
```

Comment:
casting to `Address` and then immediatelly using it as if it were of type `Address | undefined` seems like a mistake

Reply:
Yeah, I will split the `undefined` check and walletAddress cast